### PR TITLE
Added basic git hooks

### PR DESCRIPTION
**Pull request by erikhallaraaker**
This PR creates three different Git Hooks targeted at the web project.
The Git Hooks are as following:

* `post-checkout`
* `post-merge`
* `post-rewrite`

These hooks look for changes in `package-lock.json`.

If there are any changes to the packages: Install the new packages.

Most of the code is stolen shamelessly from @VikingTristan

### CODE REVIEW

File: .githooks/functions.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    if [[ $(git diff HEAD@{1}..HEAD@{0} -- "$1" | wc -l) -gt 0 ]]; then
+
+        echo -e "======> Changes in \e[33;1m$1\e[0m was found!"
+        echo -e "======> Aligning you with current branch state..."
+
+        #Run command
+        $2
+    else
+        echo -e "======> No changes found."
+fi
```
Comment (asbjornu) [inline .githooks/functions.sh]: Missing indent?

```suggestion
    fi
```

File: README.md [hunk 2]
```diff
@@ -26,26 +26,31 @@ DesignGuide requires [Node.js](https://nodejs.org/en/) (includes npm).
...
+- Open a terminal and navigate to root of repository
+- Run `git config core.hooksPath .githooks`
+
 ## Developing
 
 ### Built with
 
-* [React](https://reactjs.org/)
-* [Less](http://lesscss.org/)
+- [React](https://reactjs.org/)
```
Comment (asbjornu) [inline README.md]: These `*` to `-` changes aren't exactly related to git hooks, but 🤷‍♂️.

Comment (erikhallaraaker) [reply README.md]: Yeah.. md-linter started whining so thought I'd hit two birds with one stone

File: .githooks/functions.sh (hunk 1 above)
Comment (erikhallaraaker) [reply .githooks/functions.sh]: Indeed

File: README.md (hunk 2 above)
Comment (VikingTristan) [reply README.md]: last ned

File: .githooks/post-checkout [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+. ./.githooks/functions.sh
+
+echo "Git Hook: 'post-checkout' is running!"
+
+check_difference "package-lock.json" "npm install"
```
Comment (VikingTristan) [inline .githooks/post-checkout]: package-lock instead of package.json.... interesting

Comment (hmosvoll) [reply .githooks/post-checkout]: Agreed, package-lock.json > package.json.

**Pull request merged**